infrastructure/resource_group_level/modules/external-api.bicep

```diff
@@ -0,0 +1,185 @@
+var additionalSettings = {
+  // Note that the config didn't have AI enabled
+  // APPLICATIONINSIGHTS_CONNECTION_STRING: applicationInsights.properties.ConnectionString
+
+  DOCKER_ENABLE_CI: 'true'
+  DOCKER_REGISTRY_SERVER_URL: '${containerRegistryName}.azurecr.io'
+  DOCKER_REGISTRY_SERVER_USERNAME: dockerRegistryServerUsername
+  LOG4J_FORMAT_MSG_NO_LOOKUPS: 'true'
+
+  WEBSITE_HTTPLOGGING_RETENTION_DAYS: '3'
+  WEBSITES_ENABLE_APP_SERVICE_STORAGE: 'false'
+}
+
+var nodeEnv = environment == 'dev' ? { NODE_ENV: 'development' } : {}
+
+var appSettings = union(settings, secureSettings, additionalSettings, additionalSecureSettings, nodeEnv)
+
+var externalApiName = 'tfs-${environment}-external-api'
+var privateEndpointName = 'tfs-${environment}-external-api'
+var applicationInsightsName = 'tfs-${environment}-external-api'
+
+resource cosmosDbAccount 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' existing = {
+  name: cosmosDbAccountName
+}
+
+resource externalApi 'Microsoft.Web/sites@2022-09-01' = {
+  name: externalApiName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  kind: 'app,linux,container'
+  properties: {
+    httpsOnly: false
+    serverFarmId: appServicePlanId
+    siteConfig: {
+      numberOfWorkers: 1
+      linuxFxVersion: 'DOCKER|${dockerImageName}'
+      acrUseManagedIdentityCreds: false
+      alwaysOn: true
+      http20Enabled: true
+      functionAppScaleLimit: 0
+      // non-default parameter
+      logsDirectorySizeLimit: 100 // default is 35
+      // The following Fields have been added after comparing with dev
+      vnetRouteAllEnabled: true
+      ftpsState: 'FtpsOnly'
+      scmMinTlsVersion: '1.2'
+      remoteDebuggingVersion: 'VS2019'
+      httpLoggingEnabled: true // false in staging, true in prod
+    }
+    virtualNetworkSubnetId: appServicePlanEgressSubnetId
+  }
+}
+
+resource dtfsCentralApiSettings 'Microsoft.Web/sites/config@2022-09-01' = {
+  parent: externalApi
+  name: 'appsettings'
+  properties: appSettings
+}
+
+// The private endpoint is taken from the cosmosdb/private-endpoint export
+resource dtfsCentralApiPrivateEndpoint 'Microsoft.Network/privateEndpoints@2022-11-01' = {
+  name: privateEndpointName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  properties: {
+    privateLinkServiceConnections: [
+      {
+        name: privateEndpointName
+        properties: {
+          privateLinkServiceId: externalApi.id
+          groupIds: [
+            'sites'
+          ]
+          privateLinkServiceConnectionState: {
+            status: 'Approved'
+            actionsRequired: 'None'
+          }
+        }
+      }
+    ]
+    manualPrivateLinkServiceConnections: []
+    subnet: {
+      id: privateEndpointsSubnetId
+    }
+    ipConfigurations: []
+    // Note that the customDnsConfigs array gets created automatically and doesn't need setting here.
+  }
+}
+
+// Note that Application Insights is disabled for this resource.
```

Comment:
Currently, I'm trying to replicate the current infrastructure exactly.

I'd imagine we probably want Application Insights enabled for all our webapps, though. Is it worth the effort of preserving the differences and then harmonising them later? (Possibly - I'm not sure of the right answer here!)

Anyway, I thought it reasonable to leave this commented out to make it easier to see how it could be commonised as per FN-712.

Reply:
for now we can leave it as is it. the APIs will be refactored behind the scenes. But in the long run all apps would have App Insights enabled